tests/test_s3.py

```diff
@@ -107,6 +107,14 @@ def test_get_upload_info_max_content_length(self, app, client):
     def assert_app_config_content_length_range(self, conditions):
         assert get_condition(conditions, 'content-length-range') == [0, 100]
 
+    def test_delete_many_empty_list(self, annex, monkeypatch):
+        def assert_not_called(*args, **kwargs):
+            assert(False)
+
+        monkeypatch.setattr(annex._client, 'delete_objects', assert_not_called)
```

Comment:
why not just use a `mock` and actually assert it's not called, instead of this thing?